src/client_features.rs

```diff
@@ -247,14 +250,20 @@ impl Ord for Variant {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Debug, Clone, Eq)]
 #[cfg_attr(feature = "openapi", derive(ToSchema))]
 #[serde(rename_all = "camelCase")]
 pub struct Segment {
     pub id: i32,
     pub constraints: Vec<Constraint>,
 }
 
+impl PartialEq for Segment {
+    fn eq(&self, other: &Self) -> bool {
+        self.id == other.id
+    }
+}
```

Comment:
Deriving would include the vec of constraints as well

Reply:
The derive will respect all the fields in the struct. So it would also examine the constraints, which is not what we want in this case. We don't technically need this PartialEq but since we've implemented hash here, we should keep them in sync